src/globus_sdk/experimental/session_error/session_error.py

```diff
@@ -0,0 +1,195 @@
+import copy
+import typing as t
+
+from globus_sdk.exc import GlobusError
+
+
+class GlobusSessionErrorAuthorizationParameters:
+    """
+    Represents authorization parameters that can be used to instruct a client
+    which additional authorizations are needed in order to complete a request.
+
+    :ivar session_message: A message to be displayed to the user.
+    :vartype session_message: str, optional
+
+    :ivar session_required_identities: A list of identities required for the
+        session.
+    :vartype session_required_identities: list of str, optional
+
+    :ivar session_required_policies: A list of policies required for the
+        session.
+    :vartype session_required_policies: list of str, optional
+
+    :ivar session_required_single_domain: A list of domains required for the
+        session.
+    :vartype session_required_single_domain: list of str, optional
+
+    :ivar session_required_mfa: Whether MFA is required for the session.
+    :vartype session_required_mfa: bool, optional
+
+    :ivar session_required_scopes: A list of scopes for which consent is required.
+    :vartype session_required_scopes: list of str, optional
+
+    :ivar extra_fields: A dictionary of additional fields that were provided. May
+        be used for forward/backward compatibility.
+    :vartype extra_fields: dict
+    """
+
+    SUPPORTED_FIELDS = {
+        "session_message": str,
+        "session_required_identities": list,
+        "session_required_policies": list,
+        "session_required_single_domain": list,
+        "session_required_mfa": bool,
+        "session_required_scopes": list,
```

Comment:
The original thinking behind the `authorization_parameters` field was that that would be a dictionary that could be directly mapped to a query string (e.g. using `urlencode`) that would be passed the `/v2/oauth2/authorize` to get a valid token.  `session_requires_scopes` is not a recognized query param by auth, and since this is one specified be oauth2, we don't have the luxury of changing it.   If we are going to change the names parameters in the error format from what is passed into the `authorize` request, I'd suggest we also provide a method for constructing a valid authorize request from the error itself.

Additionally, while `required_scopes` is more descriptive than `scope`, adding `session` to it is a bit misleading since scopes have no relationship to sessions at all.  The consent can be granted in a different session and still satisfy the original token's authorization needs.

Reply:
Follow up on this: We've had some conversation about where to implement this (function/method to construct an `authorize` query string from a G.A.R.E.), and we're planning to start by implementing in the CLI and then see if we can use that to as an exercise to design something generic that makes sense to live in the SDK.